docker/pulsar-all/pom.xml

```diff
@@ -125,6 +125,29 @@
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <groupId>pl.project13.maven</groupId>
+            <artifactId>git-commit-id-plugin</artifactId>
+            <version>${git-commit-id-plugin.version}</version>
+            <executions>
+              <execution>
+                <id>git-info</id>
+                <goals>
+                  <goal>revision</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <useNativeGit>true</useNativeGit>
+              <prefix>git</prefix>
+              <failOnNoGitDirectory>true</failOnNoGitDirectory>
```

Comment:
I think that we would need to support building directly from ASF provided sources .tar.gz file. In that case there is no git directory.

Reply:
Not always. We can add a profile to enable build only with Git - we're fixing a release issue now, not generally improve the build experience.

Reply:
Support build without Git can be a follow-up. Either we calculate hash from files, required a passed tag, or keep using latest.

Reply:
If you don't set it to false, it will be unable to resolve `${git.commit.id.abbrev}`.

But I use a simple way to enable it to build the image without the git directory. The image name will be like `apachepulsar/pulsar:3.1.0-no-git`. We can further improve it with another PR if you have a better idea.